### ISSUE TITLE
Error: unable to init tuf client: unable to close from file local store: open .meta/002220.ldb: too many open files

### ISSUE THREAD

**Issue by 31337Ghost** [werf/trdl#212]
Suddenly I've reproduced an error that brought me to the previous one https://github.com/werf/trdl/issues/210.

Now every time I open new terminal I see the following error:
```console
Error: unable to init tuf client: unable to close from file local store: open /Users/golovinps/.trdl/repositories/werf/.meta/002220.ldb: too many open files
```

What is the proper way of fixing that?

**Comment by alexey-igrychev** [comment]
This issue is related to the limit of file descriptors. This limit protects the system from being overrun. On macOS, the default is usually set too low (256).

It would help if you increased the limit (e.g., you can follow [this instruction](https://neo4j.com/developer/kb/setting-max-open-file-limits-on-osx/) and reboot the system after that).

**Comment by 31337Ghost** [comment]
Thank you for your answer. 

But I definitely won't do it just to make `trdl` happy. I'm not using `trdl` in special custom manner, I'm using it just to deliver `werf` to my Mac. And as a typical user of `werf` I want that tool to just work without any custom tuning for my system.

See what I have investigated — amount of files in the repo path is very close to the limit:
```console
➜  ~ ls -la /Users/golovinps/.trdl/repositories/werf/.meta/ | wc -l
     255
➜  ~ sudo launchctl limit maxfiles
	maxfiles    256            unlimited
```
<details><summary>Complete listing of files from the repo path.</summary>

```console
➜  ~ ls -la /Users/golovinps/.trdl/repositories/werf/.meta/
total 54320
drwxr-xr-x  254 golovinps  staff    8128 Apr  5 19:25 .
drwxr-xr-x    6 golovinps  staff     192 Mar 11 13:21 ..
-rw-r--r--    1 golovinps  staff    1155 Mar 11 13:21 000004.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 11 13:22 000013.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 11 13:22 000024.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 11 13:22 000039.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 11 13:23 000042.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 11 13:23 000057.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 11 13:25 000060.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 11 13:27 000069.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 11 14:14 000078.ldb
-rw-r--r--    1 golovinps  staff  100542 Mar 11 14:19 000087.ldb
-rw-r--r--    1 golovinps  staff  100542 Mar 11 14:21 000096.ldb
-rw-r--r--    1 golovinps  staff  100542 Mar 11 14:21 000105.ldb
-rw-r--r--    1 golovinps  staff  100542 Mar 11 14:50 000116.ldb
-rw-r--r--    1 golovinps  staff  100542 Mar 12 19:14 000125.ldb
-rw-r--r--    1 golovinps  staff  100548 Mar 12 19:17 000134.ldb
-rw-r--r--    1 golovinps  staff  100548 Mar 12 19:32 000143.ldb
-rw-r--r--    1 golovinps  staff  100548 Mar 12 20:02 000152.ldb
-rw-r--r--    1 golovinps  staff  100529 Mar 12 21:04 000167.ldb
-rw-r--r--    1 golovinps  staff  100529 Mar 12 22:42 000176.ldb
-rw-r--r--    1 golovinps  staff  100545 Mar 12 22:50 000185.ldb
-rw-r--r--    1 golovinps  staff  100547 Mar 12 22:52 000194.ldb
-rw-r--r--    1 golovinps  staff  100545 Mar 12 23:53 000203.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 12 23:53 000218.ldb
-rw-r--r--    1 golovinps  staff  100541 Mar 13 19:03 000221.ldb
-rw-r--r--    1 golovinps  staff  100506 Mar 13 19:16 000230.ldb
-rw-r--r--    1 golovinps  staff  100506 Mar 13 23:52 000239.ldb
-rw-r--r--    1 golovinps  staff  100536 Mar 13 23:53 000248.ldb
-rw-r--r--    1 golovinps  staff  100536 Mar 13 23:54 000257.ldb
-rw-r--r--    1 golovinps  staff  100536 Mar 14 10:33 000266.ldb
-rw-r--r--    1 golovinps  staff  100542 Mar 14 17:46 000281.ldb
-rw-r--r--    1 golovinps  staff  101887 Mar 14 18:07 000290.ldb
-rw-r--r--    1 golovinps  staff  101879 Mar 14 18:32 000299.ldb
-rw-r--r--    1 golovinps  staff  101882 Mar 14 18:34 000308.ldb
-rw-r--r--    1 golovinps  staff  101879 Mar 14 18:34 000317.ldb
-rw-r--r--    1 golovinps  staff  101879 Mar 14 20:13 000326.ldb
-rw-r--r--    1 golovinps  staff  101881 Mar 15 10:13 000335.ldb
-rw-r--r--    1 golovinps  staff  101898 Mar 15 10:16 000344.ldb
-rw-r--r--    1 golovinps  staff  101898 Mar 15 10:51 000353.ldb
-rw-r--r--    1 golovinps  staff  101898 Mar 15 10:52 000362.ldb
-rw-r--r--    1 golovinps  staff  101898 Mar 15 11:32 000371.ldb
-rw-r--r--    1 golovinps  staff  101877 Mar 15 16:03 000380.ldb
-rw-r--r--    1 golovinps  staff  101869 Mar 15 18:33 000389.ldb
-rw-r--r--    1 golovinps  staff  103244 Mar 15 19:54 000398.ldb
-rw-r--r--    1 golovinps  staff  103244 Mar 15 21:31 000407.ldb
-rw-r--r--    1 golovinps  staff  103244 Mar 15 21:38 000416.ldb
-rw-r--r--    1 golovinps  staff  103244 Mar 15 21:38 000425.ldb
-rw-r--r--    1 golovinps  staff  103244 Mar 15 23:47 000434.ldb
-rw-r--r--    1 golovinps  staff  103251 Mar 16 00:03 000443.ldb
-rw-r--r--    1 golovinps  staff  103242 Mar 16 15:21 000452.ldb
-rw-r--r--    1 golovinps  staff  103234 Mar 16 16:01 000461.ldb
-rw-r--r--    1 golovinps  staff  103224 Mar 16 21:07 000470.ldb
-rw-r--r--    1 golovinps  staff  103239 Mar 16 21:19 000479.ldb
-rw-r--r--    1 golovinps  staff  103236 Mar 17 10:58 000488.ldb
-rw-r--r--    1 golovinps  staff  103227 Mar 17 10:59 000497.ldb
-rw-r--r--    1 golovinps  staff  103227 Mar 17 10:59 000512.ldb
-rw-r--r--    1 golovinps  staff  103227 Mar 17 11:50 000515.ldb
-rw-r--r--    1 golovinps  staff  103227 Mar 17 14:32 000524.ldb
-rw-r--r--    1 golovinps  staff  103242 Mar 17 16:57 000533.ldb
-rw-r--r--    1 golovinps  staff  103243 Mar 17 17:00 000542.ldb
-rw-r--r--    1 golovinps  staff  103243 Mar 17 17:00 000555.ldb
-rw-r--r--    1 golovinps  staff  103243 Mar 17 17:00 000560.ldb
-rw-r--r--    1 golovinps  staff  103243 Mar 17 17:00 000569.ldb
-rw-r--r--    1 golovinps  staff  103243 Mar 18 00:22 000578.ldb
-rw-r--r--    1 golovinps  staff  104554 Mar 18 10:24 000587.ldb
-rw-r--r--    1 golovinps  staff  104563 Mar 18 11:02 000596.ldb
-rw-r--r--    1 golovinps  staff  104553 Mar 19 11:52 000605.ldb
-rw-r--r--    1 golovinps  staff  104547 Mar 19 21:27 000614.ldb
-rw-r--r--    1 golovinps  staff  104558 Mar 19 21:29 000623.ldb
-rw-r--r--    1 golovinps  staff  104558 Mar 20 18:20 000632.ldb
-rw-r--r--    1 golovinps  staff  104554 Mar 20 18:26 000641.ldb
-rw-r--r--    1 golovinps  staff  104554 Mar 20 18:32 000650.ldb
-rw-r--r--    1 golovinps  staff  104554 Mar 20 20:26 000659.ldb
-rw-r--r--    1 golovinps  staff  104538 Mar 20 20:33 000668.ldb
-rw-r--r--    1 golovinps  staff  104538 Mar 20 20:48 000677.ldb
-rw-r--r--    1 golovinps  staff  104538 Mar 20 20:51 000686.ldb
-rw-r--r--    1 golovinps  staff  104538 Mar 20 23:39 000695.ldb
-rw-r--r--    1 golovinps  staff  104541 Mar 20 23:43 000704.ldb
-rw-r--r--    1 golovinps  staff  104542 Mar 21 11:08 000713.ldb
-rw-r--r--    1 golovinps  staff  104575 Mar 21 11:15 000722.ldb
-rw-r--r--    1 golovinps  staff  104575 Mar 21 12:10 000731.ldb
-rw-r--r--    1 golovinps  staff  104550 Mar 21 17:14 000740.ldb
-rw-r--r--    1 golovinps  staff  104559 Mar 21 18:03 000749.ldb
-rw-r--r--    1 golovinps  staff  104552 Mar 21 18:42 000758.ldb
-rw-r--r--    1 golovinps  staff  104552 Mar 21 20:47 000767.ldb
-rw-r--r--    1 golovinps  staff  104565 Mar 21 21:34 000776.ldb
-rw-r--r--    1 golovinps  staff  107294 Mar 21 21:40 000785.ldb
-rw-r--r--    1 golovinps  staff  107294 Mar 21 21:48 000794.ldb
-rw-r--r--    1 golovinps  staff  107294 Mar 21 21:48 000803.ldb
-rw-r--r--    1 golovinps  staff  107294 Mar 21 21:48 000812.ldb
-rw-r--r--    1 golovinps  staff  107294 Mar 21 21:50 000821.ldb
-rw-r--r--    1 golovinps  staff  107294 Mar 21 21:50 000830.ldb
-rw-r--r--    1 golovinps  staff  107294 Mar 21 21:50 000839.ldb
-rw-r--r--    1 golovinps  staff  107294 Mar 22 10:39 000848.ldb
-rw-r--r--    1 golovinps  staff  107287 Mar 22 13:49 000857.ldb
-rw-r--r--    1 golovinps  staff  107269 Mar 22 16:13 000866.ldb
-rw-r--r--    1 golovinps  staff  107268 Mar 22 17:21 000875.ldb
-rw-r--r--    1 golovinps  staff  107274 Mar 22 17:23 000884.ldb
-rw-r--r--    1 golovinps  staff  107274 Mar 22 17:30 000893.ldb
-rw-r--r--    1 golovinps  staff  107274 Mar 22 18:04 000902.ldb
-rw-r--r--    1 golovinps  staff  107279 Mar 23 09:21 000911.ldb
-rw-r--r--    1 golovinps  staff  107275 Mar 23 09:21 000920.ldb
-rw-r--r--    1 golovinps  staff  107264 Mar 23 09:21 000929.ldb
-rw-r--r--    1 golovinps  staff  107264 Mar 23 10:19 000938.ldb
-rw-r--r--    1 golovinps  staff  107279 Mar 23 10:45 000947.ldb
-rw-r--r--    1 golovinps  staff  107279 Mar 23 14:34 000956.ldb
-rw-r--r--    1 golovinps  staff  107271 Mar 23 17:29 000965.ldb
-rw-r--r--    1 golovinps  staff  107272 Mar 23 19:05 000974.ldb
-rw-r--r--    1 golovinps  staff  107257 Mar 23 19:21 000983.ldb
-rw-r--r--    1 golovinps  staff  107257 Mar 23 20:55 000992.ldb
-rw-r--r--    1 golovinps  staff  110011 Mar 24 11:21 001001.ldb
-rw-r--r--    1 golovinps  staff  109968 Mar 24 18:03 001010.ldb
-rw-r--r--    1 golovinps  staff  110017 Mar 24 18:12 001019.ldb
-rw-r--r--    1 golovinps  staff  110018 Mar 25 10:35 001028.ldb
-rw-r--r--    1 golovinps  staff  110016 Mar 25 12:01 001037.ldb
-rw-r--r--    1 golovinps  staff  110042 Mar 25 15:12 001046.ldb
-rw-r--r--    1 golovinps  staff  110031 Mar 25 15:19 001055.ldb
-rw-r--r--    1 golovinps  staff  110031 Mar 25 15:19 001064.ldb
-rw-r--r--    1 golovinps  staff  110031 Mar 25 15:27 001073.ldb
-rw-r--r--    1 golovinps  staff  110031 Mar 25 15:27 001082.ldb
-rw-r--r--    1 golovinps  staff  110031 Mar 25 15:29 001091.ldb
-rw-r--r--    1 golovinps  staff  110031 Mar 25 22:22 001100.ldb
-rw-r--r--    1 golovinps  staff  110015 Mar 26 10:26 001109.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 11:12 001118.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 11:12 001127.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 11:13 001136.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 11:22 001145.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 11:22 001154.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 11:29 001163.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 13:01 001172.ldb
-rw-r--r--    1 golovinps  staff  110013 Mar 26 13:09 001181.ldb
-rw-r--r--    1 golovinps  staff  110013 Mar 26 13:50 001190.ldb
-rw-r--r--    1 golovinps  staff  110013 Mar 26 14:03 001199.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 14:52 001208.ldb
-rw-r--r--    1 golovinps  staff  110032 Mar 26 15:52 001217.ldb
-rw-r--r--    1 golovinps  staff  110049 Mar 26 19:37 001226.ldb
-rw-r--r--    1 golovinps  staff  110047 Mar 26 19:37 001235.ldb
-rw-r--r--    1 golovinps  staff  110047 Mar 26 19:51 001244.ldb
-rw-r--r--    1 golovinps  staff  110047 Mar 26 20:51 001253.ldb
-rw-r--r--    1 golovinps  staff  110023 Mar 26 22:29 001262.ldb
-rw-r--r--    1 golovinps  staff  110028 Mar 26 22:30 001271.ldb
-rw-r--r--    1 golovinps  staff  110028 Mar 26 22:35 001280.ldb
-rw-r--r--    1 golovinps  staff  110028 Mar 27 16:03 001289.ldb
-rw-r--r--    1 golovinps  staff  110028 Mar 27 16:15 001298.ldb
-rw-r--r--    1 golovinps  staff  110028 Mar 27 18:47 001307.ldb
-rw-r--r--    1 golovinps  staff  110016 Mar 27 18:48 001316.ldb
-rw-r--r--    1 golovinps  staff  110016 Mar 27 19:01 001325.ldb
-rw-r--r--    1 golovinps  staff  110034 Mar 27 19:05 001334.ldb
-rw-r--r--    1 golovinps  staff  110034 Mar 27 19:17 001343.ldb
-rw-r--r--    1 golovinps  staff  110034 Mar 28 10:51 001352.ldb
-rw-r--r--    1 golovinps  staff  110031 Mar 28 12:23 001361.ldb
-rw-r--r--    1 golovinps  staff  110034 Mar 28 12:23 001370.ldb
-rw-r--r--    1 golovinps  staff  110034 Mar 28 18:00 001379.ldb
-rw-r--r--    1 golovinps  staff  109998 Mar 28 21:22 001388.ldb
-rw-r--r--    1 golovinps  staff  110025 Mar 28 21:58 001397.ldb
-rw-r--r--    1 golovinps  staff  110021 Mar 28 23:55 001406.ldb
-rw-r--r--    1 golovinps  staff  110023 Mar 28 23:59 001415.ldb
-rw-r--r--    1 golovinps  staff  110018 Mar 29 00:12 001424.ldb
-rw-r--r--    1 golovinps  staff  110018 Mar 29 10:22 001433.ldb
-rw-r--r--    1 golovinps  staff  110036 Mar 29 10:23 001442.ldb
-rw-r--r--    1 golovinps  staff  110036 Mar 29 10:24 001451.ldb
-rw-r--r--    1 golovinps  staff  110036 Mar 29 11:04 001460.ldb
-rw-r--r--    1 golovinps  staff  109880 Mar 29 12:49 001469.ldb
-rw-r--r--    1 golovinps  staff  109895 Mar 29 12:51 001478.ldb
-rw-r--r--    1 golovinps  staff  109895 Mar 29 13:18 001487.ldb
-rw-r--r--    1 golovinps  staff  109887 Mar 29 14:27 001496.ldb
-rw-r--r--    1 golovinps  staff  109914 Mar 29 15:38 001505.ldb
-rw-r--r--    1 golovinps  staff  109865 Mar 29 15:39 001514.ldb
-rw-r--r--    1 golovinps  staff  109863 Mar 29 15:48 001523.ldb
-rw-r--r--    1 golovinps  staff  109863 Mar 29 15:49 001532.ldb
-rw-r--r--    1 golovinps  staff  109873 Mar 29 16:10 001541.ldb
-rw-r--r--    1 golovinps  staff  109898 Mar 29 16:10 001550.ldb
-rw-r--r--    1 golovinps  staff  109898 Mar 29 19:54 001559.ldb
-rw-r--r--    1 golovinps  staff  109877 Mar 30 00:03 001568.ldb
-rw-r--r--    1 golovinps  staff  109887 Mar 30 01:14 001577.ldb
-rw-r--r--    1 golovinps  staff  109877 Mar 30 13:24 001586.ldb
-rw-r--r--    1 golovinps  staff  111310 Mar 30 13:43 001595.ldb
-rw-r--r--    1 golovinps  staff  111413 Mar 30 13:49 001604.ldb
-rw-r--r--    1 golovinps  staff  111413 Mar 30 13:54 001613.ldb
-rw-r--r--    1 golovinps  staff  111413 Mar 30 14:03 001622.ldb
-rw-r--r--    1 golovinps  staff  111407 Mar 30 14:03 001631.ldb
-rw-r--r--    1 golovinps  staff  111407 Mar 30 14:04 001640.ldb
-rw-r--r--    1 golovinps  staff  111407 Mar 30 14:20 001649.ldb
-rw-r--r--    1 golovinps  staff  111407 Mar 30 15:50 001658.ldb
-rw-r--r--    1 golovinps  staff  111409 Mar 30 16:10 001667.ldb
-rw-r--r--    1 golovinps  staff  111399 Mar 30 16:13 001676.ldb
-rw-r--r--    1 golovinps  staff  111399 Mar 30 17:53 001685.ldb
-rw-r--r--    1 golovinps  staff  111414 Mar 30 17:54 001694.ldb
-rw-r--r--    1 golovinps  staff  111414 Mar 30 19:16 001703.ldb
-rw-r--r--    1 golovinps  staff  111404 Mar 30 19:41 001712.ldb
-rw-r--r--    1 golovinps  staff  111404 Mar 30 20:16 001721.ldb
-rw-r--r--    1 golovinps  staff  111414 Mar 30 23:17 001730.ldb
-rw-r--r--    1 golovinps  staff  111417 Mar 30 23:23 001739.ldb
-rw-r--r--    1 golovinps  staff  111417 Mar 31 10:16 001748.ldb
-rw-r--r--    1 golovinps  staff  111402 Mar 31 10:30 001757.ldb
-rw-r--r--    1 golovinps  staff  111400 Mar 31 10:41 001766.ldb
-rw-r--r--    1 golovinps  staff  111400 Mar 31 11:49 001775.ldb
-rw-r--r--    1 golovinps  staff  111415 Mar 31 11:56 001784.ldb
-rw-r--r--    1 golovinps  staff  111415 Mar 31 12:24 001793.ldb
-rw-r--r--    1 golovinps  staff  111406 Mar 31 12:28 001802.ldb
-rw-r--r--    1 golovinps  staff  111406 Mar 31 13:04 001811.ldb
-rw-r--r--    1 golovinps  staff  111405 Mar 31 13:14 001820.ldb
-rw-r--r--    1 golovinps  staff  111405 Mar 31 13:19 001829.ldb
-rw-r--r--    1 golovinps  staff  111405 Mar 31 13:38 001838.ldb
-rw-r--r--    1 golovinps  staff  111405 Mar 31 13:50 001847.ldb
-rw-r--r--    1 golovinps  staff  111405 Mar 31 15:09 001856.ldb
-rw-r--r--    1 golovinps  staff  111415 Mar 31 15:23 001865.ldb
-rw-r--r--    1 golovinps  staff  111412 Mar 31 15:57 001874.ldb
-rw-r--r--    1 golovinps  staff  111412 Mar 31 16:24 001883.ldb
-rw-r--r--    1 golovinps  staff  111402 Mar 31 16:25 001892.ldb
-rw-r--r--    1 golovinps  staff  111400 Mar 31 17:53 001901.ldb
-rw-r--r--    1 golovinps  staff  111420 Mar 31 18:06 001910.ldb
-rw-r--r--    1 golovinps  staff  111409 Mar 31 18:14 001919.ldb
-rw-r--r--    1 golovinps  staff  111409 Mar 31 21:05 001928.ldb
-rw-r--r--    1 golovinps  staff  111422 Mar 31 21:08 001937.ldb
-rw-r--r--    1 golovinps  staff  111422 Mar 31 21:09 001946.ldb
-rw-r--r--    1 golovinps  staff  111422 Mar 31 21:26 001955.ldb
-rw-r--r--    1 golovinps  staff  111422 Mar 31 21:28 001964.ldb
-rw-r--r--    1 golovinps  staff  111422 Mar 31 22:26 001973.ldb
-rw-r--r--    1 golovinps  staff  111411 Mar 31 22:34 001982.ldb
-rw-r--r--    1 golovinps  staff  111411 Mar 31 23:40 001991.ldb
-rw-r--r--    1 golovinps  staff  111408 Mar 31 23:41 002000.ldb
-rw-r--r--    1 golovinps  staff  111408 Mar 31 23:42 002009.ldb
-rw-r--r--    1 golovinps  staff  111408 Mar 31 23:52 002018.ldb
-rw-r--r--    1 golovinps  staff  111408 Apr  1 09:53 002027.ldb
-rw-r--r--    1 golovinps  staff  111410 Apr  1 09:55 002036.ldb
-rw-r--r--    1 golovinps  staff  111410 Apr  1 09:58 002045.ldb
-rw-r--r--    1 golovinps  staff  111410 Apr  1 10:32 002054.ldb
-rw-r--r--    1 golovinps  staff  111408 Apr  1 10:33 002063.ldb
-rw-r--r--    1 golovinps  staff  111408 Apr  1 14:53 002072.ldb
-rw-r--r--    1 golovinps  staff  112779 Apr  2 10:56 002081.ldb
-rw-r--r--    1 golovinps  staff  112793 Apr  2 12:15 002090.ldb
-rw-r--r--    1 golovinps  staff  112799 Apr  2 12:16 002099.ldb
-rw-r--r--    1 golovinps  staff  112799 Apr  2 15:24 002108.ldb
-rw-r--r--    1 golovinps  staff  112798 Apr  2 15:42 002117.ldb
-rw-r--r--    1 golovinps  staff  112798 Apr  4 09:28 002126.ldb
-rw-r--r--    1 golovinps  staff  112789 Apr  4 17:46 002135.ldb
-rw-r--r--    1 golovinps  staff  115634 Apr  4 17:47 002144.ldb
-rw-r--r--    1 golovinps  staff  115634 Apr  4 23:57 002153.ldb
-rw-r--r--    1 golovinps  staff  115634 Apr  5 08:43 002162.ldb
-rw-r--r--    1 golovinps  staff  115647 Apr  5 08:44 002171.ldb
-rw-r--r--    1 golovinps  staff  115647 Apr  5 10:30 002180.ldb
-rw-r--r--    1 golovinps  staff  115635 Apr  5 10:54 002189.ldb
-rw-r--r--    1 golovinps  staff  115635 Apr  5 11:47 002198.ldb
-rw-r--r--    1 golovinps  staff  115632 Apr  5 11:59 002207.ldb
-rw-r--r--    1 golovinps  staff  115642 Apr  5 12:08 002228.ldb
-rw-r--r--    1 golovinps  staff  115632 Apr  5 14:26 002241.ldb
-rw-r--r--    1 golovinps  staff       0 Apr  5 19:25 002260.log
-rw-r--r--    1 golovinps  staff      16 Apr  5 19:25 CURRENT
-rw-r--r--    1 golovinps  staff      16 Apr  5 19:25 CURRENT.bak
-rw-r--r--    1 golovinps  staff       0 Mar 11 13:21 LOCK
-rw-r--r--    1 golovinps  staff  836632 Apr  5 19:25 LOG
-rw-r--r--    1 golovinps  staff   11832 Apr  5 19:25 MANIFEST-002261
```
</details>

If it's ok for `trdl` to keep so many files opened — might it shouldn't do it, according to the default ulimit that users won't tune? 
Or maybe there is some bug in the code that prevents descriptors from closing, or something like that?

**Comment by AndrewP-GH** [comment]
Same problem. Any fixes?

**Comment by alexey-igrychev** [comment]
Currently, there is no other workaround, and you should increase the limit of descriptors in your system. 
We will try to stop using leveldb as storage in the following releases.

**Comment by alexey-igrychev** [comment]
most likely fixed in v0.6.* (current stable)
